### DISCUSSION TITLE
Private Property

### DISCUSSION THREAD

**Discussion by LangSensei** [dotnet-foundation/Home#72] [General Discussions]
Is it really a bad usage of defining "private property" in class?

I know, private property doesn't sound to make sense since we should only define private fields, and define public property to expose things.

However, due to the runtime optimization, a inline property has the similar execution speed as a private field. By using only property in classes can make our codes more unified and consistent.

Wouldn't it be a good habit of always using property?

**Comment by jnm2** [comment]
I use private properties now to enable `this with { X = x }` syntax only within the class that defines `X`, since `with` doesn't work on readonly fields: https://github.com/dotnet/csharplang/discussions/5019 (I'm hoping to create a proposal soon that could change this, though.)

Another reason I've used private properties was for lazy caching.

Just like I don't put much stock in rules that say "you can never do X" (do whatever works!), I'm similarly unimpressed by rules that say "always do X." The "unified and consistent" goal feels dangerously close to the latter to me, personally.
